src/Context/product-context.js

```diff
@@ -0,0 +1,24 @@
+import { createContext } from "react";
+import { useReducer } from "react";
+import { useContext } from "react";
+
+const ProductContext = createContext({ item: 5 });
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "setProducts":
```

Comment:
In reducers, the best practise is to have the action.type in uppercase. You can change that and make the rest of the changes accordingly